pyresample/bilinear/__init__.py

```diff
@@ -0,0 +1,518 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017
+
+# Author(s):
+
+#   Panu Lahtinen <panu.lahtinen@fmi.fi>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Code for resampling using bilinear algorithm for irregular grids.
+
+The algorithm is taken from
+
+http://www.ahinson.com/algorithms_general/Sections/InterpolationRegression/InterpolationIrregularBilinear.pdf
+
+"""
+
+import numpy as np
+from pyproj import Proj
+
+from pyresample import kd_tree
+
+
+def resample_bilinear(data, in_area, out_area, radius=50e3,
+                      neighbours=32, nprocs=1, fill_value=0):
+    """Resample using bilinear interpolation.
+
+    data : numpy array
+        Array of single channel data points or
+        (source_geo_def.shape, k) array of k channels of datapoints
+    in_area : object or 2-tuple of numpy arrays
+        Geometry definition of source data or tuple of (lons, lats)
+    out_area : object
+        Geometry definition of target area
+    radius : float, optional
+        Cut-off distance in meters
+    neighbours : int, optional
+        Number of neighbours to consider for each grid point when
+        searching the closest corner points
+    nprocs : int, optional
+        Number of processor cores to be used for getting neighbour info
+    fill_value : {int, None}, optional
+            Set undetermined pixels to this value.
+            If fill_value is None a masked array is returned
+            with undetermined pixels masked
+    """
+
+    # Calculate the resampling information
+    t__, s__, input_idxs, idx_ref = get_bil_info(in_area, out_area,
+                                                 radius=radius,
+                                                 neighbours=neighbours,
+                                                 nprocs=nprocs,
+                                                 masked=False)
+
+    num_dsets = 1
+    # Handle multiple datasets
+    if data.ndim > 2 and data.shape[0] * data.shape[1] == input_idxs.shape[0]:
+        num_dsets = data.shape[2]
+        data = data.reshape(data.shape[0] * data.shape[1], data.shape[2])
+    # Also ravel single dataset
+    elif data.shape[0] != input_idxs.size:
+        data = data.ravel()
+
+    if num_dsets > 1:
```

Comment:
Why not make data 2d in any case (with second dimension shape = 1 if necessary, with newaxis or expand_dims) so that the `if` and `else` can be dropped ?